137 location module add default neutral option

gracemyz opened the pull request:
Implement a default neutral checkbox. When checked, add default neutral paths to the location module.

This particular request is not done:
"The locations for 'default neutral' should be similar to those for 'handedness' (right-hand / left-hand). That is, there should be a global setting option, but also each individual signer can be marked as having their 'own' default neutral specification. So, when a signer is selected in Sign-Level Information (this hasn't been implemented at all yet), the location module will inherit the specification from the individual."

kvesik commented:
@gracemyz When I create a new location module, it doesn't matter whether I set the articulator to Hand 1, Hand 2, or Both hands-- in all cases when I push the button to Apply 'neutral' settings, it sets the 2-handed setting (chest/breast area). 

Specific context: in settings, I changed the "default neutral two-handed location" to be body-anchored (specifically, chest/breast area) and left the "default neutral one-handed location" as is (spatial / ipsi-mid-med).

gracemyz commented:
@kvesik The 1h/2h setting is determined by the Sign type setting (if not specified, the default is two-handed). We're not actually checking which articulator is specified at all.

There's a possible conflict where Sign type is specified as 1h, but then articulator is specified as "Both hands". Currently we allow this to happen without giving any warnings.  Should we put in some checks for this? Maybe it should be addressed in a separate issue?

kvesik commented:
Oh right! Thank you @gracemyz for that reminder. But yeah, would be curious to know what @kchall thinks about that possible conflict between sign type and articulator selection.

kchall commented:
@gracemyz , @kvesik Looks great! Could we add one other checkbox option in the "Preferences / Location" dialog box, where it has the checkbox for "Automatically select 'This location is neutral' when "Apply neutral settings" button is pressed" that says "Automatically select 'This location is neutral' when "Default neutral location" is selected." (And then implement this -- "Default neutral location" is a list option under the "Purely spatial" signing space options, as shown below. I do think we should keep this as yet another way for someone to designate an underspecified neutral space instead of deleting it now that we have the other options, but it should tie into this functionality.)

<img width="777" alt="image" src="https://github.com/user-attachments/assets/68ac4890-ebb1-44eb-ac87-30abe7593e4f">

<img width="1124" alt="image" src="https://github.com/user-attachments/assets/613f2cad-73e1-457c-8156-3fd6637a879c">

kchall commented:
@gracemyz Oh wait -- actually, on further inspection, something isn't quite working right. If I specify in "Preferences" that the default neutral location should be a body-anchored location (such as the Head / Face / contra half), and then click on "Apply neutral settings," I do not get the new body-anchored location (in my case, I get the last signing-space location I tested). 

<img width="699" alt="image" src="https://github.com/user-attachments/assets/961a8baa-bcda-42ba-9145-57eb04630cd6">

gracemyz commented:
@kchall Please take a look at the updated branch! I've added the new autocheck option and an info box.
![image](https://github.com/user-attachments/assets/48bf31e9-9695-4e3f-a6d2-45cdb9f9f34f)
![image](https://github.com/user-attachments/assets/4f5b7d62-fcf9-4c26-9a3e-f08eea583c55)

gracemyz commented:
Sorry about that - I introduced some bugs when I merged from main. 
Changed to Preferences>Location, and also updated to include Kaili's location images.

kvesik commented:
@gracemyz @kchall updated default neutral behaviour looks good to me!

One concern about layout-- on my screen, the location module is now too tall to fit on the screen in its entirety, and since it can only be dragged via the top bar, it's difficult to move it into a position where the buttons at the bottom are visible. Maybe tomorrow at the meeting we can discuss options to make the layout a bit less vertical? I can think of a few off the top of my head...

Option 1:
(For all modules) move the "phonetic" location checkbox so it's next to the "phonological" one instead of below.
<img width="568" alt="image" src="https://github.com/user-attachments/assets/d385df4e-444e-4b60-ab18-43218fa8b8a9">

Option 2:
(For just the location module) rearrange the default neutral buttons onto two rows instead of three.
<img width="623" alt="image" src="https://github.com/user-attachments/assets/8e22d2d3-574a-4cdd-b826-2ef73056f828">

Option 3:
(For just the location module) reduce the amount of space that the bottom panel is allowed to take (reduce scale of image as well as max heights of location list & location details table).
![image](https://github.com/user-attachments/assets/553a5c4f-1d6f-4ae1-8a0a-23397fcc4e8c)

kchall commented:
@gracemyz This all looks great -- I think it's ready to merge in to Main, thank you!
